src/strategies/candy-lock-token/index.ts

```diff
@@ -0,0 +1,166 @@
+    lockCountMulti.call(address, contract_address, 'lockCount', [address])
+  );
+  const lockCountsResult: Record<string, BigNumber> =
+    await lockCountMulti.execute();
+  const lockCounts = Object.entries(lockCountsResult).map(
+    ([address, lockCount]) => [address, parseFloat(formatUnits(lockCount, 0))]
+  );
+
+  const lockDataMulti = new Multicaller(network, provider, v1Abi, { blockTag });
+  lockCounts.forEach(([address, lockCount]) =>
+    lockDataMulti.call(address, contract_address, 'viewLocks', [
+      address,
+      0,
+      lockCount
+    ])
+  );
+  const lockDatasMulti: Record<string, any> = await lockDataMulti.execute();
+
+  return Object.entries(lockDatasMulti).map(([address, userLocks]) => {
+    const userLockedTokenAmount = userLocks
+      .filter((userLock: V1LockData) => !userLock.unlocked)
+      .reduce(
+        (cur, acc: V1LockData) =>
+          cur + parseFloat(formatUnits(acc.amount, token_decimals)),
+        0
+      );
+    return [address, userLockedTokenAmount];
+  });
+}
+
+async function v2_3_scores(
+  network,
+  provider,
+  addresses,
+  snapshot,
+  contract_address,
+  token_decimals
+): Promise<[string, number][]> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const lockCountMulti = new Multicaller(network, provider, v2Abi, {
+    blockTag
+  });
+  addresses.forEach((address) =>
+    lockCountMulti.call(address, contract_address, 'userAllLockCount', [
+      address
+    ])
+  );
+  const lockCountsResult: Record<string, BigNumber> =
+    await lockCountMulti.execute();
+  const lockCounts = Object.entries(lockCountsResult).map(
+    ([address, lockCount]) => [address, parseFloat(formatUnits(lockCount, 0))]
+  );
+
+  const lockDataMulti = new Multicaller(network, provider, v2Abi, { blockTag });
+  lockCounts.forEach(([address, lockCount]) =>
+    lockDataMulti.call(address, contract_address, 'userAllLocks', [
+      address,
+      0,
+      lockCount
+    ])
+  );
+  const lockDatasMulti: Record<string, any> = await lockDataMulti.execute();
+
+  return Object.entries(lockDatasMulti).map(([address, userLocks]) => {
+    const userLockedTokenAmount = userLocks.reduce(
+      (cur, acc: V2LockData) =>
+        cur + parseFloat(formatUnits(acc.tokenAmount, token_decimals)),
+      0
+    );
+    return [address, userLockedTokenAmount];
+  });
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const v1Scores = await v1_scores(
+    network,
+    provider,
+    addresses,
+    snapshot,
+    options.v1_address,
+    options.decimals
+  );
+  const v2Scores = await v2_3_scores(
+    network,
+    provider,
+    addresses,
+    snapshot,
+    options.v2_address,
+    options.decimals
+  );
+  const v3Scores = await v2_3_scores(
+    network,
+    provider,
+    addresses,
+    snapshot,
+    options.v3_address,
+    options.decimals
+  );
```

Comment:
Looks like they are using `2` multicalls each, which makes it `6` multicalls, but we have a limit of `5` multicalls per strategy https://github.com/snapshot-labs/snapshot-strategies?tab=readme-ov-file#code

Reply:
I got it @ChaituVR 
Let me update it!